File: scrapers/images/loc.pl
{
	searchURL => 'http://www.loc.gov/pictures/search/?sp=%s&co=jpd&st=grid',
	searchStart => 1,
	searchEnd => 27,
	searchIncrement => 1,

	urlExtractor => scraper {
		process '//div//a[contains(@href, "collection/jpd/item")]', 'urls[]' => '@href';	
	},

	extractPageID => qr/(\d+)/,

	pageExtractor => scraper {
		process '//li[contains(text(), "Title Translation:")]/span', title => 'TEXT';
		process '//li[contains(text(), "Creator(s):")]//a', artist => 'TEXT';
		process '//li[contains(text(), "Date Created")]/span', date => 'TEXT';
		process '//a[contains(@href, "v.jpg")]', source_image => '@href';
		process '//li[contains(text(), "Summary:")]/span', description => 'TEXT';
	},

	extractImageFile => qr/([^\/]+.jpg)/
}

File: scrapers/extractedimage/loc.js
exports.extract = {
	title: '//li[contains(text(), "Title Translation:")]/span || //li[contains(text(), "Title:")]/span',
	artist: [ '//li[contains(text(), "Creator(s):")]//a || //li[contains(text(), "Related Names:")]//a', artist => artist.split( /,\s*/ ).slice( 0, -1 ).join( " " ) ],
	date: '//li[contains(text(), "Date Created")]/span',
	source_image: '//link[contains(@href, "v.jpg")]/@href',
	description: '//li[contains(text(), "Summary:")]/span',
};

exports.genURL = id => "http://www.loc.gov/pictures/collection/jpd/item/" + id + "/"
